app/src/main/java/com/codesoom/assignment/application/AuthenticationService.java

```diff
@@ -0,0 +1,75 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.domain.User;
+import com.codesoom.assignment.domain.UserRepository;
+import com.codesoom.assignment.dto.UserLoginData;
+import com.codesoom.assignment.errors.LoginInconsistencyException;
+import com.codesoom.assignment.errors.UserNotFoundException;
+import com.codesoom.assignment.errors.UnauthorizedException;
+import com.codesoom.assignment.utils.JwtUtil;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.security.SignatureException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AuthenticationService {
+
+    private final JwtUtil jwtUtil;
+    private final UserRepository userRepository;
+
+    public AuthenticationService(JwtUtil jwtUtil, UserRepository userRepository) {
+        this.jwtUtil = jwtUtil;
+        this.userRepository = userRepository;
+    }
+
+    /**
+     * 로그인을 검증합니다.
+     * @param loginData 로그인 데이터
+     * @return 로그인에 성공할 경우 토큰발급
+     */
+    public String login(UserLoginData loginData) {
+
+        User user = findUser(loginData);
+
+        if(user.getPassword().equals(loginData.getPassword())) {
+            return jwtUtil.encode(user.getId());
+        } else {
+            throw new LoginInconsistencyException();
+        }
+
+    }
```

Comment:
사용자를 찾은 후 비밀번호가 일치하면 토큰을 발행하고, 그렇지 않다면 예외를 던지는 것 같습니다. 즉 이 함수의 진짜 의도는 `토큰을 발행하는 것`인데 발행할 수 없는 경우에는 함수를 빠르게 종료하도록 하고, 제일 아래에 진짜 의도 코드를 위치하여 의도를 드러내는 것은 어떨까요?

```suggestion
    public String login(UserLoginData loginData) {
        User user = findUser(loginData);
        if(!user.getPassword().equals(loginData.getPassword())) {
            throw new LoginInconsistencyException();
        } 
        
        return jwtUtil.encode(user.getId());
    }
```

See also
  - https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html

Reply:
이러한 구문 처리를 저번에도 피드백을 해주셨는데 습관이 되어버린 것 같습니다 ㅜㅜ 더욱 명확한 의도가 나오도록 if문을 작성할때 한번 더 생각하고 작성해보도록 하겠습니다.